### DISCUSSION TITLE
Question: Using db scheduler in unit/integration tests

### DISCUSSION THREAD

**Discussion by pmcnicol** [kagkarlsson/db-scheduler#817] [General]
Hi,

First, thanks for the work on db scheduler, it’s an awesome project and working out very well for our use so far.

I was wondering if you have any recommendations on how to use db scheduler in unit/integration tests? Our preferred method of working is small spring boot micro services, where we employ @SpringBootTest style tests to test the whole micro service (eg: mocking external services, but otherwise testing web/service/db in full). 

Should we mock the Scheduler? Is there a way to enable the Scheduler but disable any execution of tasks? We’d like to be able to test scheduling a task and it’s predicted next execution, but not actually execute it (if that makes sense). 

Thanks again for the project, it’s been a pleasure to work with so far.

Patrick

**Comment by kagkarlsson** [comment]
Hi, much appreciated :). 

It is probably not perfect, but you may be able to use the `ManualScheduler` as I do in some tests: 

https://github.com/kagkarlsson/db-scheduler/blob/master/db-scheduler/src/test/java/com/github/kagkarlsson/scheduler/functional/DeleteUnresolvedTest.java#L51
